src/tetris/biasEngine/biasEvent.ts

```diff
@@ -1,10 +1,24 @@
 /// <reference path="../../../definitions/phaser.d.ts"/>
 
 import BiasEventType from "tetris/biasEngine/biasEventType"
+import BiasEventGenerator from "tetris/biasEngine/biasEventGenerator";
+import BiasEngine from "tetris/biasEngine/biasEngine";
 
-export default class BiasEvent {
+export default abstract class BiasEvent {
 
 	//region public members
+	public get minDurationInMs(): number {
```

Comment:
Yes, we used `CAPITALIZED_SNAKE_CASE` for "conventional" constants so far.
However, these "semi-constants" (i dont know if there is any better wording for this) are used differently.
The main difference between the two approaches is the later usage of the constant.

The conventional approach (`static SNAKE_CASE`) is used, when the constant is made for one class only or, when other classes exactly know where the constant is defined.
For example: When looking for the maximum value for a positive bias, i know for sure, that i need to use `BiasEngine.MAX_POSITIVE_BIAS`.

In our case at hand, this is not the case, because the `BiasEvent` is an abstract super class for many concrete `BiasEvents` (such as `DuplicateInputBiasEvent`) to inherit from it.
As such, defining a "true constant" for the minimum and maximum duration of such an event is rather difficult (because it may be too obvious if we blocked the user's input for about 5 seconds, while it might be less obvious to drop a few frames for a couple of minutes at max).
Therefore, i'm using the Prototype-Pattern within the `BiasEventGeneator` to allow subclasses of `BiasEvent` to override the (instance-bound rather than class-bound) `min/maxDurationInMs`.
This also comes with the benefit, that the `BiasEventGenerator` does not need to know the actual class of the `BiasEvent` it is currently generating (which would be needed in order to find the right constant), but purely rely on the instance function, that every `BiasEvent` has.

Reply:
Nicely explained 👍

Reply:
Thanks for the detailed explanation :). Makes sense for me now.